### PR TITLE
Add ggp init command

### PR DESCRIPTION
**Pull request by antonrohr**
The class GgpClient can now execute the command ggp ssh init -s --instance <instance_id>. This command provides the information that is needed to connect to the instance via ssh.

### CODE REVIEW

**Comment by pierricgimmig** [comment]
Hi @antonrohr , are you still working on this PR?  I see there are no reviewers, did you forget to add some?

**Comment by antonrohr** [comment]
ah yes I forgot to add them, I was late last night

File: OrbitGgp/include/OrbitGgp/GgpSshInfo.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBITGGP_GGP_SSH_INFO_H_
+#define ORBITGGP_GGP_SSH_INFO_H_
+
+#include <QJsonDocument>
```
Comment (beckerhe) [inline OrbitGgp/include/OrbitGgp/GgpSshInfo.h:8]: This include is not needed here, right? But instead we should include QByteArray.

File: OrbitGgp/GgpClient.cpp [hunk 2]
```diff
@@ -13,7 +13,62 @@
 #include "OrbitBase/Logging.h"
 #include "OrbitGgp/GgpInstance.h"
 
-const int InstanceRequestTimeoutInMilliseconds = 10'000;
+namespace {
+const int default_timeout_in_ms = 10'000;
```
Comment (dimitry-) [inline OrbitGgp/GgpClient.cpp:18]: kDefault.... and constexpr

File: OrbitGgp/GgpSshInfo.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+#include "OrbitBase/Logging.h"
+
+std::optional<GgpSshInfo> GgpSshInfo::CreateFromJson(const QByteArray& json) {
+  const QJsonDocument doc = QJsonDocument::fromJson(json);
+
+  if (!doc.isObject()) return {};
+  const QJsonObject obj = doc.object();
+
+  GgpSshInfo ggp_ssh_info;
```
[the remark this answers is not in the document]
Comment (antonrohr) [reply OrbitGgp/GgpSshInfo.cpp]: done

File: OrbitGgp/GgpSshInfo.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  if (host_val.isUndefined() || !host_val.isString() ||
+      key_path_val.isUndefined() || !key_path_val.isString() ||
+      known_hosts_path_val.isUndefined() || !known_hosts_path_val.isString() ||
+      port_val.isUndefined() || !port_val.isString() ||
+      user_val.isUndefined() || !user_val.isString()) {
+    return {};
+  }
+
+  bool ok;
+  int port = port_val.toString().toInt(&ok);
```
[the remark this answers is not in the document]
Comment (antonrohr) [reply OrbitGgp/GgpSshInfo.cpp:35]: I left a comment and added it to the test.

File: OrbitGgp/include/OrbitGgp/GgpSshInfo.h (hunk 1 above)
Comment (antonrohr) [reply OrbitGgp/include/OrbitGgp/GgpSshInfo.h]: true

File: OrbitGgp/GgpClient.cpp (hunk 2 above)
Comment (antonrohr) [reply OrbitGgp/GgpClient.cpp]: done

**Pull request merged**